src/test/java/com/vinyl/vinylstore/LoginTest.java

```diff
@@ -0,0 +1,129 @@
+package com.vinyl.vinylstore;
+
+import java.time.LocalDate;
+
+import com.vinyl.repository.TokenRepository;
+import com.vinyl.service.validation.ValidatorFactory;
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import com.vinyl.model.Address;
+import com.vinyl.model.Role;
+import com.vinyl.model.Token;
+import com.vinyl.model.User;
+import com.vinyl.modelDTO.EmailPassDTO;
+import com.vinyl.modelDTO.TokenDTO;
+
+import static org.mockito.ArgumentMatchers.anyString;
+
+public class LoginTest extends com.vinyl.vinylstore.Test {
+
+	@Autowired
+	private TokenRepository tokenRepository;
+
+	@Autowired
+	private BCryptPasswordEncoder passwordEncoder;
+
+	@Autowired
+	private ValidatorFactory validatorFactory;
+
+	private BCryptPasswordEncoder mockedEncoder;
+
+	private User user;
+
+
+	@Before
+	public void setUp() {
+		user = createUser();
+		mockedEncoder = Mockito.mock(BCryptPasswordEncoder.class);
+		Mockito.when(mockedEncoder.matches(anyString(), anyString())).thenReturn(true);
+		validatorFactory.setPasswordEncoder(mockedEncoder);
+	}
+
+	@After
+	public void tearDown() {
+		tokenRepository.deleteAll();
+		userRepository.deleteAll();
+		addressRepository.deleteAll();
+		roleRepository.deleteAll();
+		validatorFactory.setPasswordEncoder(passwordEncoder);
+	}
+
+	@Test
+	public void loginWithValidEmailAndPassword() {
```

Comment:
this test is for loginWithValidEmailAndPass and when token already exists, rename it as such and add a new one when token does not exists and it will be created when the request is done

Reply:
done